File: apps/CardHero.NetCoreApp.TypeScript/ClientApp/models/DeckModel.ts
import IMapper from "../utils/mapper";
import DeckCardModel from "./DeckCardModel";

export interface DeckId extends Number {
    _deckIdBrand: number;
}

export default class DeckModel implements IMapper<DeckModel> {
    id: DeckId;
    name: string;
    description?: string;
    maxCards: number;
    isFavourited: boolean;
    cards: DeckCardModel[];

    from(o?: any): DeckModel {
        if (!o) return this;

        this.description = o.description;
        this.id = o.id;
        this.isFavourited = o.isFavourited;
        this.maxCards = o.maxCards;
        this.name = o.name;

        if (o.cards) {
            this.cards = o.cards.map(x => new DeckCardModel().from(x));
        }

        return this;
    }
}

File: src/CardHero.Core.Models/Models/DeckModel.cs
using System.Collections.Generic;

namespace CardHero.Core.Models
{
    /// <summary>
    /// Deck.
    /// </summary>
    public class DeckModel : ICardHeroEntity
    {
        /// <summary>
        /// Id.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Name.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Description.
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Max cards.
        /// </summary>
        public int MaxCards { get; set; }

        /// <summary>
        /// Is favourited.
        /// </summary>
        public bool IsFavourited { get; set; }

        /// <summary>
        /// List of cards.
        /// </summary>
        public IEnumerable<DeckCardModel> Cards { get; set; }
    }
}
